### DISCUSSION TITLE
Is it possible to extract word/sentence from vector?

### DISCUSSION THREAD

**Discussion by saahiluppal** [explosion/spaCy#5349] [Help: Coding & Implementations] [answered]
can we convert vector back to it's corresponding word in spaCy?

**Comment by adrianeboyd** [comment, marked as answer]
You can use [Vectors.most_similar](https://spacy.io/api/vectors#most_similar) to find the word(s) closest to a given vector. In a loaded model, you can access the vectors under `nlp.vocab.vectors`, so:

```
# convert vector to numpy array (if it isn't already)
query_vector = numpy.asarray([0.1, 0.3, 0.5, ...])
keys, _, _ = nlp.vocab.vectors.most_similar([query_vector])
# get string for the most similar hash
print(nlp.vocab.strings[keys[0][0]])
```

(I just noticed that the example in the docs uses `nlp.vectors` instead of `nlp.vocab.vectors`. We'll get that updated.)
